app/assets/javascripts/study_overview/study-overview.js

```diff
@@ -0,0 +1,287 @@
+/**
+* @fileoverview Default view of Explore tab in Study Overview
+*
+* Shows "Clusters" and sometimes "Genomes", etc.
+*/
+
+const study = window.SCP.study
+
+window.SCP.startPendingEvent('user-action:page:view:site-study',
+  { speciesList: window.SCP.taxons },
+  'plot:',
+  true)
+
+/**
+ * Render cluster code
+ *
+ * TODO: Move this to render-cluster.js, or some such
+ */
+
+/** Get Plotly layout object for scatter plot */
+function getBaseLayout(height, width, font) {
+  const layout = {
+    hovermode: 'closest',
+    margin: {
+      t: 25,
+      r: 0,
+      b: 20,
+      l: 0
+    },
+    height,
+    width,
+    font
+  }
+  return layout
+}
+
+/** Gets Plotly layout scene props for 3D scatter plot */
+function get3DScatterProps(camera, cluster) {
+  const { clusterHasRange, axes } = cluster
+  const { titles, ranges, aspects } = axes
+
+  const scene = {
+    camera,
+    aspectmode: 'cube',
+    xaxis: { title: titles.x, autorange: true, showticklabels: false },
+    yaxis: { title: titles.y, autorange: true, showticklabels: false },
+    zaxis: { title: titles.z, autorange: true, showticklabels: false }
+  }
+
+  if (clusterHasRange) {
+    scene.xaxis.autorange = false
+    scene.xaxis.range = ranges.x
+    scene.yaxis.autorange = false
+    scene.yaxis.range = ranges.y
+    scene.zaxis.autorange = false
+    scene.zaxis.range = ranges.x
+    scene.aspectmode = aspects.mode,
+    scene.aspectratio = {
+      x: aspects.x,
+      y: aspects.y,
+      z: aspects.z
+    }
+  }
+
+  return scene
+}
+
+/** Gets Plotly layout props for 2D scatter plot */
+function get2DScatterProps(cluster) {
+  const {
+    axes, clusterHasRange, domainRanges, hasCoordinateLabels, coordinateLabels
+  } = cluster
+  const { titles } = axes
+
+  const layout = {
+    xaxis: { title: titles.x, showticklabels: false },
+    yaxis: { title: titles.y, showticklabels: false, scaleanchor: 'x' }
+  }
+
+  // if user has supplied a range, set that, otherwise let Plotly autorange
+  if (clusterHasRange) {
+    layout.xaxis.range = domainRanges.x
+    layout.yaxis.range = domainRanges.y
+  } else {
+    layout.xaxis.autorange = true
+    layout.yaxis.autorange = true
+  }
+
+  if (hasCoordinateLabels) {
+    layout.annotations = coordinateLabels
+  }
+
+  return layout
+}
+
+/** Calculate the current viewport to use for rendering cluster plots */
+function calculatePlotViewport(numPlots) {
+  const windowDom = $(window)
+  const plotHeight = windowDom.height() - 250
+  const plotWidth = (windowDom.width() - 80) / numPlots
+  return { plotHeight, plotWidth }
+}
+
+/** Renders Plotly scatter plot for "Clusters" tab */
+function drawScatterPlot(data, is3D, height, width, labelFont) {
```

Comment:
> multiple targets

`drawScatterPlot` has one target: `#plots .panel-body`.

> hardcoded incrementing seems unusual

It enables unique plot IDs, per [comment at initial assignment](https://github.com/broadinstitute/single_cell_portal_core/pull/782/files/0117dd3703bb498093cecf5d645769b8d3c59561#diff-7e1476b83dde40a39e49d7383ec9914d5cfb565b8324c047e3807db048c4100fR27).  That seems like a simple, robust solution.  Could you help me understand what's unusual about it?

Reply:
right, sorry I didn't express what I meant well.  I'm not concerned about the  strategy for ensuring unique ids.  What I'm surprised at is that we have a 'draw' function that always just tacks on the plot at the end of the container, rather than accepting an argument for where to draw the plot.   So, for example, if you want to render two plots, with one plot always on the left, it looks like you have to invoke the draws in a specific order, which might not correspond to when the plot data becomes available.

 The pattern that I would rather see (and that I think is more common) is that the logic for "how to draw a plot" (e.g. the plotly invocation and camera setup) is separate from the logic of deciding where on the page to plot it (the latter being the logic that I typically think of as 'application controller' logic. )

Reply:
Gotcha, that makes sense!  I'll refactor to take a "position" (or similar) argument, to ensure that e.g. cluster plots appear at left and spatial plots at right, regardless of which data loads first.

Reply:
Cool, I'd strongly suggest that the argument is a dom element ID.  That matches e.g. renderMorpheusDotPlot(), ideoConfig, and in general is the most common pattern for 'draw' functions.  Then we can have a separate function/controller which manages the multi-plot panel (and where the eventual inter-plot interactivity code will go).

Reply:
Thanks for the motivating example.  References to [`renderMorpheusPlot`](https://github.com/broadinstitute/single_cell_portal_core/search?q=renderMorpheusDotPlot) clarify the use case for a DOM selector parameter: reuse in different views.  I'll add it here, as it's trivial and likely needed soon.  Neither of the "draw" functions -- [`drawDotplot`](https://github.com/broadinstitute/single_cell_portal_core/blob/0b6e3c132ee11ee850937edb27abf4ec5139bce5/app/assets/javascripts/scp-dot-plot.js#L247) and [`drawHeatmap`](https://github.com/broadinstitute/single_cell_portal_core/blob/be278fd0bead3cfa8d608c0a3f216091e6b08522/app/views/site/_view_gene_expression_heatmap.html.erb#L61) -- take a DOM selector, but per your suggestion we can make them if/when they support multi-plotting with spatial data.

Let's pursue your new architectural suggestion for multi-plot controllers in SCP-2857.  There we'll integrate new API endpoints and can show off your proposal's utility.  Regarding eventual inter-plot interactivity -- agreed, and it will be a separate bolus of work.  Demo feedback revealed it will have rather specific requirements, and Product noted we should defer it until after extending spatial multi-plots to gene search results.

Reply:
ah, great point on 'draw' vs. 'render'.  thanks!

Reply:
> rename my drawScatterPlot to renderScatterPlot and add a DOM selector parameter

Done in 159aca4.